app/models/issue.rb

```diff
@@ -45,4 +45,16 @@ def lt_ln_present?(obj)
   def status_name
     STATUSES[status]
   end
+
+  def published?
+    (status == 'open') || (status == 'closed')
+  end
+
+  def can_read_by?(user)
+    if user.present?
+      user.admin? || user.moderator? || (self.user == user) || published?
+    else
+      false
+    end
+  end
```

Comment:
I apologize, no need in class ```Guest```, just ```User.new``` will work, i'll update comment.
To check if user is logged in we can use ```user_signed_in?``` , it is recommended by devise: https://github.com/plataformatec/devise#controller-filters-and-helpers